ironfish/src/chainProcessor.ts

```diff
@@ -122,13 +122,14 @@ export class ChainProcessor {
       await this.add(add)
       this.hash = add.hash
       this.sequence = add.sequence
-
+      hashChanged = true
```

Comment:
If I'm reading `findFork` and `iterateTo` correctly, it is possible that `add.hash` may be the same as the initial `this.hash`. In that case, wouldn't setting `hashChanged = true` potentially get consumers stuck in an infinite loop (because they would keep calling `update()`)?

Maybe I'm wrong, because if what I'm describing is right, then some tests would fail. But by looking at the code I'm not convinced that this change is totally harmless.

Reply:
I added a test to verify this, so it should be more verifiable now via a test.